### PR TITLE
[train] high level cleanup of Ray Train docs

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

1.  Removed "Advanced" User Guides page - This is purely organizational and can be added back in the future if it makes sense.
     1. Moved the two guides underneath to be top-level.
2. Removed "Internals" page. - This grouping did not make sense.
     1. Removed "Ray Train Architecture" - This is outdated. Prefer removing this for now and adding an updated one in the future. 
     3. Removed "Environment Variables" - These are not that useful. If needed we should add these to the relevant user guide.
     4. Moved "Ray Train Benchmarks" to be top-level.
         1. Removed data ingest benchmark
3. Removed the "FAQ" - These are redundant with other pages.

New ToC:

image

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/benchmarks.rst
Comment (woshiyyya) [inline doc/source/train/benchmarks.rst:1]: Maybe off the topic, these benchmarks are too old and toy. We may need more real-world workloads to validate our performance.

**Review by woshiyyya — APPROVED** [review]
Thanks Matt! That looks good to me

File: doc/source/train/benchmarks.rst
Comment (angelinalg) [reply doc/source/train/benchmarks.rst:1]: Get rid of it if it's outdated! Yay!

**Review by angelinalg — APPROVED** [review]
Great job with your eye for pruning. Thank you for doing this!!

**Pull request merged**